plugins/modules/ds8000_volume_mapping.py

```diff
@@ -0,0 +1,207 @@
+    type: str
+  volume_name:
+    description:
+      - The volume name that you want to map to a host.
+      - Notice that different volumes have the same volume name, so it will map all of them.
+      - To use a specific volume, use C(volume_id)
+    type: str
+extends_documentation_fragment:
+  - ibm.ds8000.ds8000.documentation
+'''
+
+EXAMPLES = r'''
+- name: Ensure that a host is exists on the storage
+  ibm.ds8000.ds8000_volume_mapping:
+    hostname: "{{ ds8000_host }}"
+    username: "{{ ds8000_username }}"
+    password: "{{ ds8000_password }}"
+    name: host_name_test
+    state: present
+
+- name: Ensure that a host does not exists on the storage
+  ibm.ds8000.ds8000_volume_mapping:
+    hostname: "{{ ds8000_host }}"
+    username: "{{ ds8000_username }}"
+    password: "{{ ds8000_password }}"
+    name: host_name_test
+    state: absent
+'''
+
+RETURN = r''' # '''
+
+from ..module_utils.ds8000 import (PyDs8k, costume_get_request,
+                                   ds8000_argument_spec)
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+import json
+
+
+class PyDs8kHelper(PyDs8k):
+    def __init__(self, module):
+        super(PyDs8kHelper, self).__init__(module)
+        self.changed = False
+        self.failed = False
+
+    def _ds8000_volume_mapping_present(self, volume_id):
+        result = {}
+        changed = False
+        failed = False
+        volume_mappinng_on_host = self._get_volume_mapping_on_specific_host()
+        for volume_map in volume_mappinng_on_host:
+            if volume_id == volume_map['volume']['id']:
+                result = {'changed': changed, 'failed': failed}
+                return result
+        changed, failed = self._map_volume_to_host(volume_id)
+        result = {'changed': changed, 'failed': failed}
+        return result
+
+    def _map_volume_to_host(self, volume_id):
+        name = self.params['name']
+        client = self.client
+        changed = False
+        failed = False
+        try:
+            client.map_volume_to_host(host_name=name,
+                                      volume_id=volume_id)
+            changed = True
+        except Exception as generic_exc:
+            failed = True
+            self.module.fail_json(
+                msg="Failed to map volume id {volume_id} to {host_name} host on DS8000 storage. "
+                "ERR: {error}".format(
+                    volume_id=volume_id,
+                    host_name=name,
+                    error=to_native(generic_exc)))
+        return changed, failed
+
+    def _ds8000_volume_mapping_absent(self, volume_id):
+        result = {}
+        changed = False
+        failed = False
+        volume_mappinng_on_host = self._get_volume_mapping_on_specific_host()
+        for volume_map in volume_mappinng_on_host:
+            if volume_id == volume_map['volume']['id']:
+                changed, failed = self._unmap_volume_from_host(volume_map)
```

Comment:
Let's try to create a third function that does this logic since it's very similar to the `present` function.